sycl/doc/extensions/experimental/sycl_ext_oneapi_device_global.asciidoc

```diff
@@ -227,17 +227,24 @@ class device_global {
 public:
   using element_type = std::remove_extent_t<T>; 
   
+#if !__cpp_consteval
   static_assert(std::is_trivially_default_constructible_v<T>,
                 "Type T must be trivially default constructable (until C++20 "
                 "consteval is supported and enabled)");
+#endif // !__cpp_consteval
 
   static_assert(std::is_trivially_destructible_v<T>,
       "Type T must be trivially destructible.");
 
-  // Only default construction is allowed.  The underlying memory allocations
-  // of type T on devices will be zero-initialized before any non-initialization
-  // accesses occur.
-  device_global();
+  // The underlying memory allocations of type T on devices will be 
+  // zero-initialized before any non-initialization accesses occur.
+  device_global() = default;
+
+  // device_global initializes underlying T with the args argument
+#if __cpp_consteval
+  template <typename... Args>
+  consteval explicit device_global(Args&&... args);
+#endif // __cpp_consteval
```

Comment:
Don't we want this instead?

```
#if __cpp_consteval
  // device_global initializes underlying T with the args argument
  template <typename... Args>
  consteval explicit device_global(Args&&... args);
#else
  static_assert(std::is_trivially_default_constructible_v<T>,
                  "Type T must be trivially default constructable (until C++20 "
                  "consteval is supported and enabled)");

  // The underlying memory allocations of type T on devices will be 
  // zero-initialized before any non-initialization accesses occur.
  device_global() = default;
#endif // __cpp_consteval
```

This ensures that one of the following is true:

* The type `T` has a `constexpr` constructor with the given parameters.  (If the parameter pack is empty, this invokes the default constructor.)  Or,
* The type `T` has a trivial default constructor.

The way you have it written now, a user with a C++20 compiler could have a `T` with a default constructor that is neither trivial nor `constexpr`.  I don't think we want to allow that case because we would not know the initial value at compile-time.

Reply:
You are correct, updated to reflect this.